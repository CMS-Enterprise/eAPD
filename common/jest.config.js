export default {
  coverageDirectory: './coverage-endpoint/',
  rootDir: './',
  testMatch: ['**/*.test.{js,jsx}'],
  transform: {
    '\\.[jt]sx?$': 'babel-jest'
  },
  testEnvironment: 'node',
  testTimeout: 30000,
  collectCoverageFrom: ['**/*.{js,jsx}'],
  testPathIgnorePatterns: ['/node_modules/', '/coverage-endpoint/'],
  coverageThreshold: {
    global: {
      branches: 74,
      functions: 80,
      lines: 80
    }
  }
};
