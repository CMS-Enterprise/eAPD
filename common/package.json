{
  "name": "@cms-eapd/common",
  "version": "0.0.1",
  "type": "module",
  "description": "APD app for CMS HITECH program",
  "exports": "./index.js",
  "scripts": {
    "lint": "eslint '**/*.js'",
    "stryker": "TZ=utc stryker run",
    "test": "DEBUG_PRINT_LIMIT=10000 TZ=utc jest",
    "test-watch": "DEBUG_PRINT_LIMIT=10000 TZ=utc jest --watch",
    "npm-check": "npx npm-check-updates"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/CMSgov/eAPD.git"
  },
  "license": "CC0-1.0",
  "bugs": {
    "url": "https://github.com/CMSgov/eAPD/issues"
  },
  "homepage": "https://github.com/CMSgov/eAPD#readme",
  "engines": {
    "node": "16.16.0"
  },
  "dependencies": {
    "@joi/date": "2.1.0",
    "joi": "17.6.0"
  },
  "devDependencies": {
    "@stryker-mutator/core": "6.1.2",
    "@stryker-mutator/jest-runner": "6.1.2"
  },
  "packageManager": "yarn@1.22.18"
}
