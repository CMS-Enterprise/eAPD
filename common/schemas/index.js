// Combined schema is used for validating a full APD object
export { default as combinedSchemas } from './combined';

export { default as activitiesDashboard } from './activitiesDashboard';
export {
  default as activityOverview,
  activitySummary,
  activityDescription
} from './activityOverview';
export { default as apdOverview } from './apdOverview';
export { default as assurancesAndCompliance } from './assurancesAndCompliance';
export {
  default as costAllocationFFP,
  activityCostAllocationFFP
} from './costAllocationFFP';
export {
  default as costAllocationOther,
  activityCostAllocationOther
} from './costAllocationOther';
export { default as costAllocateFFPQuarterly } from './costAllocateFFPQuarterly';
export { default as costAllocation } from './costAllocation';
export {
  default as incentivePayments,
  proposedBudgetEhAmt,
  proposedBudgetEpAmt
} from './incentivePayments';
export {
  default as keyMedicaid,
  medicaidDirector,
  medicaidOffice
} from './keyMedicaid';
export { default as keyPerson } from './keyPerson';
export { default as milestones } from './milestones';
export {
  default as nameAndFundingSource,
  activityName,
  activityFundingSource
} from './nameAndFundingSource';
export { default as nonPersonnelCosts } from './nonPersonnelCosts';
export { default as outcomeMetric } from './outcomeMetric';
export { default as personCost } from './personCost';
export {
  default as plannedActivitySchedule,
  activityStartDate,
  activityEndDate
} from './plannedActivityShedule';
export { default as privateContractor } from './privateContractor';
export { default as standardsAndConditions } from './standardsAndConditions';
export { default as statePersonnel } from './statePersonnel';
