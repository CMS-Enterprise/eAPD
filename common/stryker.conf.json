{
  "dryRunTimeoutMinutes": 30,
  "timeoutMS": 500000,
  "mutate": ["utils/budget.js"],
  "$schema": "./node_modules/@stryker-mutator/core/schema/stryker-schema.json",
  "testRunner": "jest",
  "reporters": ["progress", "clear-text", "html"],
  "coverageAnalysis": "off",
  "jest": {
    "projectType": "custom",
    "configFile": "jest.config.js",
    "config": {
      "testEnvironment": "node"
    },
    "enableFindRelatedTests": true
  }
}
