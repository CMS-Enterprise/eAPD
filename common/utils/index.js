export { default as roundedPercents } from './roundedPercents.js';

export * from './auth.js';
export * from './budget.js';
export * from './formatting.js';
export * from './utils.js';
export * from './constants.js';
export * from './test-data/exampleMMISBudget.js';
