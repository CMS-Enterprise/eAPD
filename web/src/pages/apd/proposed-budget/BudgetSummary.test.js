import { shallow } from 'enzyme';
import React from 'react';

import { APD_TYPE } from '@cms-eapd/common';

import {
  plain as BudgetSummary,
  mapStateToProps,
  DataRow,
  DataRowGroup,
  HeaderRow
} from './BudgetSummary';

const sampleBudgetMMIS = {
  ddi: {
    '90-10': {
      keyStatePersonnel: {
        2023: {
          total: 2400075,
          federal: 2124939,
          medicaid: 2361044,
          state: 236104
        },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: {
          total: 2400075,
          federal: 2124939,
          medicaid: 2361044,
          state: 236104
        }
      },
      statePersonnel: {
        2023: {
          total: 2400075,
          federal: 2124939,
          medicaid: 2361044,
          state: 236104
        },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: {
          total: 2400075,
          federal: 2124939,
          medicaid: 2361044,
          state: 236104
        }
      },
      contractors: {
        2023: {
          total: 1982756,
          federal: 1746939,
          medicaid: 1941043,
          state: 194104
        },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: {
          total: 1982756,
          federal: 1746939,
          medicaid: 1941043,
          state: 194104
        }
      },
      expenses: {
        2023: {
          total: 775000,
          federal: 675670,
          medicaid: 750744,
          state: 75075
        },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: {
          total: 775000,
          federal: 675670,
          medicaid: 750744,
          state: 75075
        }
      },
      combined: {
        2023: {
          total: 5157831,
          federal: 4547548,
          medicaid: 5052831,
          state: 505283
        },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: {
          total: 5157831,
          federal: 4547548,
          medicaid: 5052831,
          state: 505283
        }
      }
    },
    '75-25': {
      keyStatePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      statePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      contractors: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      expenses: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      combined: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      }
    },
    '50-50': {
      keyStatePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      statePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      contractors: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      expenses: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      combined: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      }
    },
    combined: {
      2023: {
        total: 5157831,
        federal: 4547548,
        medicaid: 5052831,
        state: 505283
      },
      2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
      total: {
        total: 5157831,
        federal: 4547548,
        medicaid: 5052831,
        state: 505283
      }
    }
  },
  mando: {
    '75-25': {
      keyStatePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      statePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 1286801,
          federal: 965101,
          medicaid: 1286801,
          state: 321700
        },
        total: {
          total: 1286801,
          federal: 965101,
          medicaid: 1286801,
          state: 321700
        }
      },
      contractors: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 542444,
          federal: 406833,
          medicaid: 542444,
          state: 135611
        },
        total: {
          total: 542444,
          federal: 406833,
          medicaid: 542444,
          state: 135611
        }
      },
      expenses: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 75000, federal: 56250, medicaid: 75000, state: 18750 },
        total: { total: 75000, federal: 56250, medicaid: 75000, state: 18750 }
      },
      combined: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 1904245,
          federal: 1428184,
          medicaid: 1904245,
          state: 476061
        },
        total: {
          total: 1904245,
          federal: 1428184,
          medicaid: 1904245,
          state: 476061
        }
      }
    },
    '50-50': {
      keyStatePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      statePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      contractors: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      expenses: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      combined: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      }
    },
    combined: {
      2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
      2024: {
        total: 1904245,
        federal: 1428184,
        medicaid: 1904245,
        state: 476061
      },
      total: {
        total: 1904245,
        federal: 1428184,
        medicaid: 1904245,
        state: 476061
      }
    }
  },
  combined: {
    2023: {
      total: 5257831,
      federal: 4628548,
      medicaid: 5142831,
      state: 514283
    },
    2024: {
      total: 5097245,
      federal: 3792934,
      medicaid: 5047245,
      state: 1254311
    },
    total: {
      total: 10355076,
      federal: 8421482,
      medicaid: 10190076,
      state: 1768594
    }
  }
};

describe('budget summary component', () => {
  test('renders correctly for hitech apds', () => {
    const component = shallow(
      <BudgetSummary
        apdType={APD_TYPE.HITECH}
        activities={{
          hie: ['hie data'],
          hit: ['hit data'],
          mmis: ['mmis data']
        }}
        data={{
          hie: {},
          hit: {},
          mmis: {},
          combined: {
            1: { federal: 1, state: 2, medicaid: 1000, total: 3 },
            2: { federal: 10, state: 20, medicaid: 2000, total: 30 }
          }
        }}
        years={['1', '2']}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test('renders correctly for mmis apds', () => {
    const component = shallow(
      <BudgetSummary
        apdType={APD_TYPE.MMIS}
        activities={{
          mmis: ['mmis data']
        }}
        data={sampleBudgetMMIS}
        years={['2023', '2024']}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test('renders DataRowGroup correctly for mmis apds', () => {
    const component = shallow(
      <DataRowGroup
        apdType={APD_TYPE.MMIS}
        data={sampleBudgetMMIS.ddi['90-10']}
        year={'2023'}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test('data row renders with no data', () => {
    expect(
      shallow(
        <DataRow
          category="category text"
          data={{
            medicaid: 0,
            federal: 0,
            state: 0,
            total: 0
          }}
          entries={[]}
          title="title text"
          year="year"
        />
      )
    ).toMatchSnapshot();
  });

  test('data row renders with data', () => {
    const component = shallow(
      <DataRow
        category="category text"
        data={{ federal: 1, state: 2, medicaid: 1000, total: 3 }}
        entries={[]}
        title="title text"
        year="year"
      />
    );
    expect(component).toMatchSnapshot();
  });

  test('data row group renders', () => {
    expect(
      shallow(
        <DataRowGroup
          data={{
            combined: { 1448: {} },
            contractors: { 1448: {} },
            expenses: { 1448: {} },
            statePersonnel: { 1448: {} }
          }}
          entries={[10, 20, 30]}
          year="1448"
        />
      )
    ).toMatchSnapshot();
  });

  test('data row group renders with keyStatePersonnel', () => {
    expect(
      shallow(
        <DataRowGroup
          apdType={APD_TYPE.MMIS}
          data={{
            combined: { 1448: {} },
            contractors: { 1448: {} },
            expenses: { 1448: {} },
            statePersonnel: { 1448: {} },
            keyStatePersonnel: { 1448: {} }
          }}
          entries={[10, 20, 30]}
          year="1448"
        />
      )
    ).toMatchSnapshot();
  });

  test('header row renders', () => {
    expect(shallow(<HeaderRow yr="1" />)).toMatchSnapshot();
    expect(shallow(<HeaderRow yr="total" />)).toMatchSnapshot();
  });

  test('maps state to props', () => {
    const state = {
      apd: { data: { years: ['1', '2'] } },
      budget: {
        activityTotals: [
          { fundingSource: 'HIE', name: 'activity 1' },
          { fundingSource: 'HIT', name: 'activity 2' },
          { fundingSource: 'MMIS', name: 'activity 3' },
          { fundingSource: 'MMIS', name: 'activity 4' }
        ]
      }
    };

    expect(mapStateToProps(state)).toEqual({
      activities: {
        hie: [{ fundingSource: 'HIE', name: 'activity 1' }],
        hit: [{ fundingSource: 'HIT', name: 'activity 2' }],
        mmis: [
          { fundingSource: 'MMIS', name: 'activity 3' },
          { fundingSource: 'MMIS', name: 'activity 4' }
        ]
      },
      data: {
        activityTotals: [
          { fundingSource: 'HIE', name: 'activity 1' },
          { fundingSource: 'HIT', name: 'activity 2' },
          { fundingSource: 'MMIS', name: 'activity 3' },
          { fundingSource: 'MMIS', name: 'activity 4' }
        ]
      },
      years: ['1', '2']
    });
  });
});
