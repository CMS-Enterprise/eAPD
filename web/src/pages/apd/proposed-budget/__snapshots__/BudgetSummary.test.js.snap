// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`budget summary component data row group renders 1`] = `
<Fragment>
  <DataRow
    apdType="HITECH"
    category="statePersonnel"
    data={Object {}}
    key="0"
    title="State Staff"
  />
  <DataRow
    apdType="HITECH"
    category="expenses"
    data={Object {}}
    key="1"
    title="Other State Expenses"
  />
  <DataRow
    apdType="HITECH"
    category="contractors"
    data={Object {}}
    key="2"
    title="Private Contractor"
  />
  <DataRow
    apdType="HITECH"
    category="combined"
    data={Object {}}
    key="3"
    title="Subtotal"
  />
</Fragment>
`;

exports[`budget summary component data row group renders with keyStatePersonnel 1`] = `
<Fragment>
  <DataRow
    apdType="MMIS"
    category="keyStatePersonnel"
    data={Object {}}
    key="0"
    title="Key State Personnel"
  />
  <DataRow
    apdType="MMIS"
    category="statePersonnel"
    data={Object {}}
    key="1"
    title="State Staff"
  />
  <DataRow
    apdType="MMIS"
    category="expenses"
    data={Object {}}
    key="2"
    title="Other State Expenses"
  />
  <DataRow
    apdType="MMIS"
    category="contractors"
    data={Object {}}
    key="3"
    title="Private Contractor"
  />
  <DataRow
    apdType="MMIS"
    category="combined"
    data={Object {}}
    key="4"
    title="Subtotal"
  />
</Fragment>
`;

exports[`budget summary component data row renders with data 1`] = `
<tr
  className="
      "
>
  <th
    className="indent-title budget-table--col-divider__right"
    scope="row"
  >
    title text
  </th>
  <td
    className="budget-table--number"
  >
    <Dollars>
      1
    </Dollars>
  </td>
  <td
    className="budget-table--number"
  >
    <Dollars>
      2
    </Dollars>
  </td>
  <td
    className="budget-table--number"
  >
    <Dollars>
      1000
    </Dollars>
  </td>
</tr>
`;

exports[`budget summary component data row renders with no data 1`] = `
<tr
  className="
      "
>
  <th
    className="indent-title budget-table--col-divider__right"
    scope="row"
  >
    title text
  </th>
  <td
    className="budget-table--number"
  >
    <Dollars>
      0
    </Dollars>
  </td>
  <td
    className="budget-table--number"
  >
    <Dollars>
      0
    </Dollars>
  </td>
  <td
    className="budget-table--number"
  >
    <Dollars>
      0
    </Dollars>
  </td>
</tr>
`;

exports[`budget summary component header row renders 1`] = `
<tr>
  <th
    id="summary-budget-fy-1-undefined"
    key="1"
  >
    FFY 1
  </th>
  <th
    className="ds-u-text-align--right"
    scope="col"
  >
    Federal Share
  </th>
  <th
    className="ds-u-text-align--right"
    scope="col"
  >
    State Share
  </th>
  <th
    className="ds-u-text-align--right"
    scope="col"
  >
    Medicaid Total Computable
  </th>
</tr>
`;

exports[`budget summary component header row renders 2`] = `
<tr>
  <th
    id="summary-budget-fy-total-undefined"
    key="total"
  >
    Total
  </th>
  <th
    className="ds-u-text-align--right"
    scope="col"
  >
    Federal Share
  </th>
  <th
    className="ds-u-text-align--right"
    scope="col"
  >
    State Share
  </th>
  <th
    className="ds-u-text-align--right"
    scope="col"
  >
    Medicaid Total Computable
  </th>
</tr>
`;

exports[`budget summary component renders correctly for hitech apds 1`] = `
<div>
  <div>
    <h4
      aria-hidden="true"
      className="ds-h4 header-with-top-margin"
    >
      HIT Activities
    </h4>
    <table
      className="budget-table"
      data-cy="summaryBudgetHIT"
      key="1"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        1
         HIT Activities
      </caption>
      <thead>
        <HeaderRow
          activity="hit"
          yr="1"
        />
      </thead>
      <tbody>
        <DataRowGroup
          data={Object {}}
          entries={
            Array [
              "hit data",
            ]
          }
          year="1"
        />
      </tbody>
    </table>
    <table
      className="budget-table"
      data-cy="summaryBudgetHIT"
      key="2"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        2
         HIT Activities
      </caption>
      <thead>
        <HeaderRow
          activity="hit"
          yr="2"
        />
      </thead>
      <tbody>
        <DataRowGroup
          data={Object {}}
          entries={
            Array [
              "hit data",
            ]
          }
          year="2"
        />
      </tbody>
    </table>
    <table
      className="budget-table"
      data-cy="summaryBudgetHIT"
      key="total"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        total
         HIT Activities
      </caption>
      <thead>
        <HeaderRow
          activity="hit"
          yr="total"
        />
      </thead>
      <tbody>
        <DataRowGroup
          data={Object {}}
          entries={
            Array [
              "hit data",
            ]
          }
          year="total"
        />
      </tbody>
    </table>
  </div>
  <div>
    <h4
      aria-hidden="true"
      className="ds-h4 header-with-top-margin"
    >
      HIE Activities
    </h4>
    <table
      className="budget-table"
      data-cy="summaryBudgetHIE"
      key="1"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        1
         HIE Activities
      </caption>
      <thead>
        <HeaderRow
          activity="hie"
          yr="1"
        />
      </thead>
      <tbody>
        <DataRowGroup
          data={Object {}}
          entries={
            Array [
              "hie data",
            ]
          }
          year="1"
        />
      </tbody>
    </table>
    <table
      className="budget-table"
      data-cy="summaryBudgetHIE"
      key="2"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        2
         HIE Activities
      </caption>
      <thead>
        <HeaderRow
          activity="hie"
          yr="2"
        />
      </thead>
      <tbody>
        <DataRowGroup
          data={Object {}}
          entries={
            Array [
              "hie data",
            ]
          }
          year="2"
        />
      </tbody>
    </table>
    <table
      className="budget-table"
      data-cy="summaryBudgetHIE"
      key="total"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        total
         HIE Activities
      </caption>
      <thead>
        <HeaderRow
          activity="hie"
          yr="total"
        />
      </thead>
      <tbody>
        <DataRowGroup
          data={Object {}}
          entries={
            Array [
              "hie data",
            ]
          }
          year="total"
        />
      </tbody>
    </table>
  </div>
  <div>
    <h4
      aria-hidden="true"
      className="ds-h4 header-with-top-margin"
    >
      MMIS Activities
    </h4>
    <table
      className="budget-table"
      data-cy="summaryBudgetMMIS"
      key="1"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        1
         MMIS Activities
      </caption>
      <thead>
        <HeaderRow
          activity="mmis"
          yr="1"
        />
      </thead>
      <tbody>
        <DataRowGroup
          apdType="HITECH"
          data={Object {}}
          entries={
            Array [
              "mmis data",
            ]
          }
          year="1"
        />
      </tbody>
    </table>
    <table
      className="budget-table"
      data-cy="summaryBudgetMMIS"
      key="2"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        2
         MMIS Activities
      </caption>
      <thead>
        <HeaderRow
          activity="mmis"
          yr="2"
        />
      </thead>
      <tbody>
        <DataRowGroup
          apdType="HITECH"
          data={Object {}}
          entries={
            Array [
              "mmis data",
            ]
          }
          year="2"
        />
      </tbody>
    </table>
    <table
      className="budget-table"
      data-cy="summaryBudgetMMIS"
      key="total"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        total
         MMIS Activities
      </caption>
      <thead>
        <HeaderRow
          activity="mmis"
          yr="total"
        />
      </thead>
      <tbody>
        <DataRowGroup
          apdType="HITECH"
          data={Object {}}
          entries={
            Array [
              "mmis data",
            ]
          }
          year="total"
        />
      </tbody>
    </table>
  </div>
  <table
    className="budget-table"
    data-cy="summaryBudgetTotals"
  >
    <caption
      className="ds-h4"
    >
      Activities Totals
    </caption>
    <thead>
      <tr>
        <td
          className="th"
          id="summary-budget-null1"
        />
        <th
          className="ds-u-text-align--right"
          scope="col"
        >
          Federal Total
        </th>
        <th
          className="ds-u-text-align--right"
          scope="col"
        >
          State Total
        </th>
        <th
          className="ds-u-text-align--right"
          scope="col"
        >
          Medicaid Total Computable
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="budget-table--row__highlight budget-table--total"
        key="1"
      >
        <th
          scope="row"
        >
          Activities Grand Total
        </th>
        <td
          className="budget-table--number budget-table--col-divider__left"
        >
          <Dollars>
            1
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars>
            2
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars>
            1000
          </Dollars>
        </td>
      </tr>
      <tr
        className=""
        key="2"
      >
        <th
          scope="row"
        >
          FFY 2
        </th>
        <td
          className="budget-table--number budget-table--col-divider__left"
        >
          <Dollars>
            10
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars>
            20
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars>
            2000
          </Dollars>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`budget summary component renders correctly for mmis apds 1`] = `
<div>
  <div>
    <h4
      aria-hidden="true"
      className="ds-h4 header-with-top-margin"
    >
      MMIS Activities
    </h4>
    <table
      className="budget-table"
      data-cy="summaryBudgetMMIS"
      key="1"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        1
         MMIS Activities
      </caption>
      <thead>
        <HeaderRow
          activity="mmis"
          yr="1"
        />
      </thead>
      <tbody>
        <DataRowGroup
          apdType="MMIS"
          data={Object {}}
          entries={
            Array [
              "mmis data",
            ]
          }
          year="1"
        />
      </tbody>
    </table>
    <table
      className="budget-table"
      data-cy="summaryBudgetMMIS"
      key="2"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        2
         MMIS Activities
      </caption>
      <thead>
        <HeaderRow
          activity="mmis"
          yr="2"
        />
      </thead>
      <tbody>
        <DataRowGroup
          apdType="MMIS"
          data={Object {}}
          entries={
            Array [
              "mmis data",
            ]
          }
          year="2"
        />
      </tbody>
    </table>
    <table
      className="budget-table"
      data-cy="summaryBudgetMMIS"
      key="total"
    >
      <caption
        className="ds-u-visibility--screen-reader"
      >
        FFY 
        total
         MMIS Activities
      </caption>
      <thead>
        <HeaderRow
          activity="mmis"
          yr="total"
        />
      </thead>
      <tbody>
        <DataRowGroup
          apdType="MMIS"
          data={Object {}}
          entries={
            Array [
              "mmis data",
            ]
          }
          year="total"
        />
      </tbody>
    </table>
  </div>
  <table
    className="budget-table"
    data-cy="summaryBudgetTotals"
  >
    <caption
      className="ds-h4"
    >
      Activities Totals
    </caption>
    <thead>
      <tr>
        <td
          className="th"
          id="summary-budget-null1"
        />
        <th
          className="ds-u-text-align--right"
          scope="col"
        >
          Federal Total
        </th>
        <th
          className="ds-u-text-align--right"
          scope="col"
        >
          State Total
        </th>
        <th
          className="ds-u-text-align--right"
          scope="col"
        >
          Medicaid Total Computable
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="budget-table--row__highlight budget-table--total"
        key="1"
      >
        <th
          scope="row"
        >
          Activities Grand Total
        </th>
        <td
          className="budget-table--number budget-table--col-divider__left"
        >
          <Dollars>
            1
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars>
            2
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars>
            1000
          </Dollars>
        </td>
      </tr>
      <tr
        className=""
        key="2"
      >
        <th
          scope="row"
        >
          FFY 2
        </th>
        <td
          className="budget-table--number budget-table--col-divider__left"
        >
          <Dollars>
            10
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars>
            20
          </Dollars>
        </td>
        <td
          className="budget-table--number"
        >
          <Dollars>
            2000
          </Dollars>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
